### PR TITLE
Apache Ignite updates from Sergey

### PR DESCRIPTION
**Pull request by DavideD**
Follows: https://github.com/DavideD/hibernate-ogm-ignite-bak/pull/2

### CODE REVIEW

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteProperties.java [hunk 1]
```diff
@@ -29,6 +32,41 @@
...
+	/**
+	 * Configuration property templates for manage read-through and write-through for each entity.
+	 * Example of full string "hibernate.ogm.ignite.persistent_store.Person.read_through=true"
+	 * @see <a href="https://apacheignite.readme.io/docs/persistent-store#read-through-and-write-through">Persistent Store</a>
+	 */
+
+	public static final String IGNITE_CACHE_STORE_CLASS_TEMPLATE = "hibernate.ogm.ignite.persistent_store.%s.cache_store_class";
+	public static final String IGNITE_CACHE_STORE_FACTORY_TEMPLATE = "hibernate.ogm.ignite.persistent_store.%s.cache_store_factory";
+
+	public static final String IGNITE_ALLOWS_TRANSACTION_EMULATION = "hibernate.ogm.ignite.allows_transaction_emulation";
```
Comment (DavideD) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteProperties.java:44]: @schernolyas This should not be an option, it should always be true or false depending if Ignite support transactions or not

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteDatastoreProvider.java [hunk 2]
```diff
@@ -435,19 +460,19 @@ private String findKeyType(EntityKeyMetadata keyMetadata) {
...
-				if ( result == null ) {
-					//if nothing found we use id field name
-					result = StringHelper.stringBeforePoint( keyMetadata.getColumnNames()[0] );
-					result = StringUtils.capitalize( result );
-				}
+				if ( result == null ) { */
+				//if nothing found we use id field name
+				result = StringHelper.stringBeforePoint( keyMetadata.getColumnNames()[0] );
+				result = StringUtils.capitalize( result );
+				//}
```
Comment (DavideD) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteDatastoreProvider.java:475]: @schernolyas Could you explain a bit the reason behind these changes?

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteDatastoreProvider.java [hunk 3]
```diff
@@ -470,4 +498,16 @@ public String getEntityTypeName(String entity) {
 	private String getEntityCacheName(String entity) {
 		return StringHelper.stringBeforePoint( entity ) ;
 	}
+
+	@Override
+	public boolean allowsTransactionEmulation() {
+		boolean allowsTransactionEmulation = propertyReader.property( IgniteProperties.IGNITE_ALLOWS_TRANSACTION_EMULATION, Boolean.class )
+				.withDefault( Boolean.FALSE ).getValue();
+		return allowsTransactionEmulation;
+	}
```
Comment (DavideD) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteDatastoreProvider.java:507]: @schernolyas  This shouldn't be an option

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/options/impl/CacheStoreFactoryOption.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.ignite.options.impl;
+
+import org.hibernate.ogm.options.spi.UniqueOption;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class CacheStoreFactoryOption extends UniqueOption<Class> {
```
Comment (DavideD) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/options/impl/CacheStoreFactoryOption.java:14]: @schernolyas  Some javadoc would be nice

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/query/impl/IgniteSqlQueryParser.java [hunk 5]
```diff
@@ -76,7 +76,7 @@ public String process() {
 	}
 
 	public IgniteQueryDescriptor buildQueryDescriptor() {
-		return new IgniteQueryDescriptor( process(), null, true /* SQL queries working only for scalars queries */ );
+		return new IgniteQueryDescriptor( process(),"==111====", null, true /* SQL queries working only for scalars queries */ );
```
Comment (DavideD) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/query/impl/IgniteSqlQueryParser.java:79]: @schernolyas Is this right?

File: ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/test/mapping/CompositeEmbeddedIdTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class CompositeEmbeddedIdTest extends OgmTestCase {
```
Comment (DavideD) [inline ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/test/mapping/CompositeEmbeddedIdTest.java:26]: @schernolyas Isn't this one the same test in core?

File: ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/test/mapping/News.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.Objects;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+@Entity
+public class News {
```
Comment (DavideD) [inline ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/test/mapping/News.java:17]: @schernolyas  You don't need to copy these classes

File: integrationtest/pom.xml [hunk 8]
```diff
@@ -155,26 +155,10 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>hibernate-ogm-ignite</artifactId>
             <scope>provided</scope>
+            <version>${project.version}</version>
```
Comment (DavideD) [inline integrationtest/pom.xml:158]: @schernolyas There is no need to add the version to these dependencies, it is inherited from the parent

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteProperties.java (hunk 1 above)
Comment (schernolyas) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteProperties.java:44]: @DavideD I think same. But ... @Z-z-z-z developed by  another way. You can see classes org.hibernate.ogm.datastore.ignite.transaction.impl.ApplicationServer  and org.hibernate.ogm.datastore.ignite.transaction.impl.IgniteTransactionManagerFactory .  Also, we have a problem with Narayana. These reasons  pushed me to use transction emulation (intergation with native transactions of Ignite). Details of it you can see  https://apacheignite.readme.io/docs/transactions

**Comment by DavideD** [comment]
OK @schernolyas, I'm going to close this one and wait for your new one

**Pull request closed without merging**